pkg/web/src/app/Providers/components/AddEditProviderModal/AddEditProviderModal.tsx

```diff
@@ -496,6 +501,32 @@ export const AddEditProviderModal: React.FunctionComponent<IAddEditProviderModal
                         </Button>
                       </Popover>
                     }
+                    labelInfo={
+                      <Button
+                        isInline
+                        icon={<ExternalLinkSquareAltIcon />}
+                        iconPosition="right"
+                        variant="link"
+                        isDisabled={!fields.hostname?.isValid}
+                        onClick={async () => {
+                          const { data } = await caCertQuery.refetch();
+                          if (data) {
+                            fields.caCert?.setValue(data.pemEncoded);
+                            fields.caCert?.setIsTouched(true);
+                            fields.caCertFilename?.setValue('pki-resource');
+                          }
+                        }}
```

Comment:
@mturley - Does this look reasonable?  It was the first way I found that made sense to run the cert fetch query and then update the contents of more that one field.

Reply:
Hmm, yeah I think this is reasonable. It's hard to say because having an imperative fetch button like this goes a little bit against the design of react-query (declaratively render loaded data, it will handle when to fetch under the hood) but that's necessary for this use case.

We may want to make sure we have a consistent UX and implementation with the way vsphere certificate fingerprints are handled (and maybe that means changing vsphere to behave more like this, bringing back the ability to paste the fingerprint). In the vsphere case, the button simply [sets boolean state](https://github.com/konveyor/forklift-ui/blob/6eb656453f271364211795ded932227bc91bbc20/pkg/web/src/app/Providers/components/AddEditProviderModal/AddEditProviderModal.tsx#L387-L389) that enables the certificate query, and when that query succeeds it [renders certificate information and an "I trust this" checkbox](https://github.com/konveyor/forklift-ui/blob/6eb656453f271364211795ded932227bc91bbc20/pkg/web/src/app/Providers/components/AddEditProviderModal/AddEditProviderModal.tsx#L398-L470). When that checkbox is changed, that's when the hidden form field is actually updated with the fingerprint data. There's also a mechanism there when editing an existing provider where it can detect that the fingerprint has changed since the provider was created and show a warning.

Maybe since we're dealing with a fingerprint for vsphere and a whole CA cert for ovirt we can't make this 100% consistent, but I'm just sharing for comparison's sake. In that case since we didn't need to fill in the form field immediately as data was loaded we could get away with using the regular query `enabled` flag to control fetch time. But if you are going to have the field visible here, since you need to fetch and then fill in the field immediately after (and you don't want some `useEffect` overwriting the field later if the user edits it) your implementation makes sense to me.

Another option to imperatively get data from react-query if you know the data has already been fetched is to get a reference to the `queryClient` via [`useQueryClient()`](https://react-query-v3.tanstack.com/reference/useQueryClient) and then call `queryClient.getQueryData(queryKey)`. Similarly not ideal, but won't cause a refetch if you know you don't need one.